ui/apps/platform/cypress/integration/vulnerabilities/workloadCves/imageSingle.test.js

```diff
@@ -0,0 +1,175 @@
+                        cy.get(`${bySeverityCard} ${selectors.iconText(severity)}`);
+                        cy.get(`${bySeverityCard} ${selectors.iconText('Results hidden')}`).should(
+                            'have.length',
+                            3
+                        );
+
+                        // Check that the filtered view label is present
+                        cy.get(selectors.filteredViewLabel);
+
+                        // Check that the row count in the header above the table matches the CVE count for the image
+                        cy.get(`*:contains("${count} results found")`);
+
+                        // Check that the count and severity in the summary card still match after the filter is applied
+                        cy.get(`${bySeverityCard} ${selectors.iconText(`${count} ${severity}`)}`);
+
+                        // Check that the total number of fixable + not fixable matches the total number of CVEs
+                        cy.get(
+                            [
+                                `${byStatusCard} ${selectors.iconText('with available fixes')}`,
+                                `${byStatusCard} ${selectors.iconText('without fixes')}`,
+                            ].join(',')
+                        ).then(([$fixable, $notFixable]) => {
+                            const fixableCount = parseInt(
+                                $fixable.innerText.replace(/\D/g, ''),
+                                10
+                            );
+                            const notFixableCount = parseInt(
+                                $notFixable.innerText.replace(/\D/g, ''),
+                                10
+                            );
+                            expect(fixableCount + notFixableCount).to.equal(count);
+                        });
+
+                        // Check that every row in the table has the correct severity
+                        cy.get(`table tbody tr td[data-label="CVE severity"]`).each(($severity) => {
+                            expect($severity.text()).to.equal(severity);
+                        });
+                    });
+            });
+    });
+
+    // This test should correctly apply a CVE name filter to the CVEs table
+    it('should correctly apply CVE name filters', () => {
+        // Visit the workload CVE overview page
+        visitWorkloadCveOverview();
+
+        // Select the image tab
+        cy.get(selectors.entityTypeToggleItem('Image')).click();
+
+        // Select any image that has CVEs in the table
+        cy.get(
+            `tbody tr:has(td[data-label="CVEs by severity"] ${selectors.nonZeroCveSeverityCounts})`
+        )
+            .first()
+            .then(([$rowWithCves]) => {
+                // Click the link in the table to visit the image page
+                cy.wrap($rowWithCves).find('td[data-label="Image"] a').click();
```

Comment:
In this test, image page does not depend on any specific information from images page (bravo).

Use of `then` suggests that assertions for second page depend on information from first page, as in the previous test.

To flatten the test, what do you think about either of the following with method or pseudo-selector:

```js
cy.get(
    `tbody tr:has(td[data-label="CVEs by severity"] ${selectors.nonZeroCveSeverityCounts})`
).first().find('td[data-label="Image"] a').click();
```

```js
cy.get(`tbody tr:has(td[data-label="CVEs by severity"] ${selectors.nonZeroCveSeverityCounts}):first td[data-label="Image"] a`).click();
```

P.S. Your discovery of `has` pseudo-selector opens the door to this possibility. Earlier tests needed `then` block to do this.

Reply:
Great catch, I wasn't thrilled about the `then` block here but didn't think to use `find`!

Reply:
Style-wise I liked the latter, but it looks like it isn't possible to chain the `:has()` and `:first` pseudo-selectors. I checked the jquery [docs](https://api.jquery.com/first-selector/) and it looks like `:first` is deprecated anyway and usage of `.first()` is preferred.

Reply:
Aha, what I meant when I wrote `:first` was `:nth(0)` although not necessarily recommending it.